pkg/prove/querier.go

```diff
@@ -54,3 +58,87 @@ func QueryTxInclusionProof(_ sdk.Context, path []string, req abci.RequestQuery)
 
 	return rawTxProof, nil
 }
+
+const ShareInclusionQueryPath = "shareInclusionProof"
+
+// QueryShareInclusionProof defines the logic performed when querying for the inclusion
+// proofs of a set of shares to the data root.
+// the shares range should be appended to the path.
+// example path for proving the set of shares [3, 5]:
+// custom/shareInclusionProof/3/5
+func QueryShareInclusionProof(_ sdk.Context, path []string, req abci.RequestQuery) ([]byte, error) {
+	// parse the shares range from the path
+	if len(path) != 2 {
+		return nil, fmt.Errorf("expected query path length: 2 actual: %d ", len(path))
+	}
+	beginShare, err := strconv.ParseInt(path[0], 10, 64)
+	if err != nil {
+		return nil, err
+	}
+	endShare, err := strconv.ParseInt(path[1], 10, 64)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
we should probably create an issue for creating code to find these easily

Reply:
What do you mean by to find these easily?

Reply:
the beginning and ending share that a message starts on